src/types.d.ts

```diff
@@ -0,0 +1,1194 @@
+   * Scene
+   */
+  export class Scene {
+    constructor(viewer: Viewer, view: View)
+
+    createLayer(opts?: {
+      source: Source
+      geometry: Geometry
+      pinFirstLevel?: boolean
+      textureStoreOpts?: TextureStoreOptions
+      layerOpts?: LayerOptions
+    }): Layer
+
+    destroy(): void
+
+    destroyAllLayers(): void
+
+    destroyLayer(layer: Layer): void
+
+    hotspotContainer(): HotpsotContainer
+
+    layer(): Layer
+
+    viewer(): Viewer
+
+    listLayers(): Layer[]
+
+    lookTo(params: LookToParams, opts?: LookToOptions, done?: Function): void
+
+    movement(): Function
+
+    startMovement(fn: Function, done?: Function): void
+
+    stopMovement(): void
+
+    switchTo(opts?: SwitchSceneOptions, done?: Function): void
+
+    view(): View
+
+    viewer(): Viewer
+
+    visibled(): boolean
+
+    addEventListener(event: Event, callback: Function)
+    removeEventListener(event: Event, callback: Function)
+  }
+
+  /**
+   * ScrollZoomControlMethod
+   */
+  export class ScrollZoomControlMethod {
+    constructor(
+      element?: HTMLElement,
+      opts?: { fictionTime?: number; zoomDelta?: number }
+    )
+
+    destroy(): void
+  }
+
+  export interface ScrollZoomControlMethodInterface
+    extends ScrollZoomControlMethod {}
+
+  /**
+   * SingleAssetSource
+   */
+  export class SingleAssetSource {
+    constructor(asset: Asset)
+
+    loadAsset(stage: Stage, tile: Tile, done?: Function): void
+  }
+
+  /**
+   * StaticAsset
+   */
+  export class StaticAsset implements Asset {
+    constructor(element: HTMLElement)
+
+    element(): HTMLElement
+
+    height(): number
+
+    isDynamic(): false
+
+    timestamp(): number
+
+    width(): number
+  }
+
+  /**
+   * TextureStore
+   */
+  export class TextureStore {
+    constructor(source: Source, stage: Stage, opts?: TextureStoreOptions)
+
+    clear(): void
+
+    clearNotPinned(): void
+
+    destroy(): void
+
+    endFrame(): void
+
+    markTile(tile: Tile): void
+
+    pin(tile: Tile): number
+
+    query(tile: Tile): any
+
+    source(): Source
+
+    stage(): Stage
+
+    startFrame(): void
+
+    unpin(tile: Tile): number
+  }
+
+  export interface TextureStoreOptions {
+    previouslyVisibleCacheSize?: number
+  }
+
+  /**
+   * TextureStore
+   */
+  export class TextureStoreItem {
+    constructor(store: TextureStore, tile: Tile)
+  }
+
+  /**
+   * TileSearcher
+   */
+  export class TileSearcher {
+    search(view: View, tile: Tile, result: Tile[]): number
+  }
+
+  /**
+   * Timer
+   */
+  export class Timer {
+    constructor(opts?: { duration?: number })
+
+    duration(): number
+
+    start(): number
+
+    started(): boolean
+
+    stop(): boolean
+  }
+
+  /**
+   * VelocityControlMethod
+   */
+  export class VelocityControlMethod {
+    constructor(parameter: string)
+
+    destroy(): boolean
+
+    setFriction(friction: number): void
+
+    setVelocity(velocity: number): void
+  }
+
+  export interface VelocityControlMethodInterface
```

Comment:
This is duplicated

Reply:
What's the purpose of the VelocityControlMethodInterface? When are we supposed to use it?